runtime/src/main/kotlin/kotlin/native/concurrent/Freezing.kt

```diff
@@ -42,12 +42,25 @@ public fun <T> T.freeze(): T {
 public val Any?.isFrozen
     get() = isFrozenInternal(this)
 
+
+/**
+ * Transforms an object to the frozen form by creating frozen version of the object, by either
+ * deep copy of mutable objects and reusing frozen references. Note, that consequent calls
+ * of this function may return different instances.
+ *
+ * @return the frozen object form
+ * @see ensureNeverFrozen
+ */
+public inline fun <reified T> T.toFrozen(): T =
+        @Suppress("UNCHECKED_CAST", "NON_PUBLIC_CALL_FROM_PUBLIC_INLINE")
```

Comment:
`@PublishedApi` can be added to `toFrozenInternal` to avoid unreliable suppression of compiler error.